Query SKUs instead of hardcoding them

sabifa opened the pull request:
Hello!
Thanks for your efforts with this script.

I noticed that NVIDIA tends to change the SKU from time to time (already happened for German drops), which causes this script to stop monitoring the stock correctly.

My change introduces a new api call to query the currently active SKUs and use them to check stock.
This also introduces a new configuration approach.
I have also added a new locale configuration to quickly switch between regions.

For now, there are still some debug printouts in the code so that you are able to verify that the correct SKU is found. Feel free to remove these or change my code to better fit your code guidelines.

Please let me know what you think!

sabifa committed:
Add sku query before checking stock

sku changed in german shop after the first drop, so in order to always have the latest sku we have to query it with the script

sabifa committed:
Remove old config

sabifa committed:
Update readme

oelcode commented:
Hey @sabifa - thanks for this!

I’m out for most of the day, so I’ve only been able to look at this briefly on my phone.

My only questions are:

1) How often are the SKU’s checked for changes? I’m reluctant to send unnecessary requests to the API.

2) Does your amendment still send an individual request for each SKU? No problem if it does, I just hadn’t gotten round to figuring out if it was possible to combine requests. 

3) Have you been able to test this with multiple regions? I’m thinking that I could create a “known locale codes” list that can be contributed to.

sabifa commented:
1.  Currently they are checked on each cycle, so yes with this implementation the amount of api calls are bumped by one. It may be a good idea to query them not that often (although I am currently running the script and haven't been blocked yet).
2. Yes it still does - in fact I didn't know that it's possible to combine them! Sounds quite useful to keep the requests as low as possible.
3. That sounds like a great idea! I've only tested en-gb (as for backwards compatibility) and de-de (the location I am monitoring) so far.

oelcode committed:
Updated 50check.py with new functionality

Your code prompted some ideas, so I rewrote a few aspects of the script and fixed a couple of things that had broken.

oelcode committed:
Updated config.py

Updated config.py to include the new/changed configuration options following the updates to the main script.

oelcode committed:
Update README.md

Updated README.md to include updated feature information.

oelcode reviewed (approved):
@sabifa - Thanks for your PR. It worked really well, so I decided to build on it and make things a little more robust.

- I've changed the SKU check to run at a configurable interval (SKU_CHECK_CONFIG) (default is 1 hour).
- In addition, I added a function that will warn the user several times if the API product list is checked and it no longer matches the configured card options.
- I haven't figured out how to combined the API requests yet, maybe that's a job for tomorrow!
- I created a quick config tool, 'stockconfig.py'. I tested a few locale codes and made a list of the ones that definitely work (I couldn't get en-us to work at the time of testing though...). This tool prompts the user for the correct locale, and to confirm which cards they want to monitor (using the latest API provided list of cards).
- I noticed that the SKU's differ slightly in some countries, so the config tool should help with making sure the correct ones are being used for that locale.
- I'm not a fan of filling the console with each check, but I appreciate it's a quick way to ensure the script is still working. I've setup a config option ("log_stock_checks":) to toggle whether it is performed. The default is False.
- Some various logging and error enhancement fixes were done.

Thanks again for your help!

sabifa commented:
Thanks for taking a look! I like your additions to the config and the limitation of the SKU cache update. 
Now the script just has to help us get a card 😄 

Happy FE hunting!
